### PR TITLE
Fix automaxprocs log

### PR DESCRIPTION
**Pull request by CrazyHZM**
# Description

## Issue reference

Fix https://github.com/dapr/dapr/issues/3895

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by daixiang0 — APPROVED** [review]
LGTM

File: cmd/daprd/main.go [hunk 1]
```diff
@@ -152,6 +152,13 @@ var (
 )
 
 func main() {
+	// set GOMAXPROCS
+	undo, err := maxprocs.Set()
+	defer undo()
+	if err != nil {
+		log.Fatal("failed to set GOMAXPROCS: %v", err)
+	}
+
```
Comment (Taction) [inline cmd/daprd/main.go:161]: Thanks for fixing it. Generally looks good to me, just `maxprocs.Set()` I think is enough, undo and error handling are not necessary. Besides, better move it to `init`.

```go
func init() {
	maxprocs.Set()
}
```

Comment (CrazyHZM) [reply cmd/daprd/main.go]: Done.

**Review by wcs1only — APPROVED** [review]

**Review by Taction — APPROVED** [review]
LGTM

**Review by berndverst — APPROVED** [review]

**Review by pkedy — APPROVED** [review]

**Pull request merged**